roles/test_operator/tasks/tempest-tests.yml

```diff
@@ -56,3 +56,53 @@
     - name: Set variable
       ansible.builtin.set_fact:
         cifmw_test_operator_tempest_exclude_list: "{{ list_skipped.skipped_tests | join('\n') }}"
+
+- name: Ensure a secret for the cifmw private key file exists
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: v1
+      kind: Secret
+      type: Opaque
+      metadata:
+        name: "{{ cifmw_test_operator_controller_priv_key_secret_name }}"
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      data:
+        ssh-privatekey: "{{ lookup('file', '~/.ssh/id_cifw') | b64encode }}"
+  when:
+    - not cifmw_test_operator_dry_run | bool
+    - cifmw_test_operator_tempest_ssh_key_secret_name is not defined
+
+- name: Add SSHKeySecretName section to Tempest CR
+  ansible.builtin.set_fact:
+    test_operator_cr: >-
+      {{
+          test_operator_cr |
+          combine({'spec': {'SSHKeySecretName':
+                    cifmw_test_operator_controller_priv_key_secret_name
+                  }}, recursive=true)
+      }}
+  when:
+    - not cifmw_test_operator_dry_run | bool
+    - cifmw_test_operator_tempest_ssh_key_secret_name is not defined
+
+- name: Add controller IP section to Tempest CR
+  ansible.builtin.set_fact:
+    test_operator_cr: >-
+      {{
+          test_operator_cr |
+          combine({'spec': {'tempestconfRun': {'overrides':
+                    (test_operator_cr.spec.tempestconfRun.overrides | default('')) + ' ' +
+                    'whitebox_neutron_plugin_options.proxy_host_address ' + controller_ip
+                  }}}, recursive=true)
+      }}
+  when:
+    - not cifmw_test_operator_dry_run | bool
+    - controller_ip != ""
+  vars:
+    controller_ip: >-
+      {{ cifmw_test_operator_controller_ip | default(ansible_default_ipv4.address) | default('') }}
```

Comment:
Cosmetic/consistency comment: We usually put then clauses and vars at the very first of the task definition. Easier to read as you can skip reading the task as soon as you know the when condition is false.

Reply:
Thank you, fixed! I'm going to stick with this rule next time I'll be proposing a PR.